Examples/Io/Csv/include/ActsExamples/Io/Csv/CsvMeasurementReader.hpp

```diff
@@ -0,0 +1,82 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#pragma once
+
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Geometry/TrackingGeometry.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "ActsExamples/Framework/IReader.hpp"
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+namespace Acts {
+class Surface;
+}
+
+namespace ActsExamples {
+
+/// Read in a measurement cluster collection in comma-separated-value format.
+///
+/// This reads three files per event file in the configured input
+/// directory. By default it reads file in the current working directory.
+/// Files are assumed to be named using the following schema
+///
+///     event000000001-cells.csv (optional)
+///     event000000001-measurements.csv
+///     event000000001-truth.csv
+///     event000000002-cells.csv (optional)
+///     event000000002-measurements.csv
+///     event000000002-truth.csv
+///
+/// and each line in the file corresponds to one hit/cluster.
+///
+/// One file per fevent: thread-safe for parallel event porcessing.
+class CsvMeasurementReader final : public IReader {
+ public:
+  struct Config {
+    /// Where to read input files from.
+    std::string inputDir;
+    /// Output measurement collection.
+    std::string outputMeasurements;
+    /// Output cluster collection (optional).
+    std::string outputClusters;
+    /// For each cluster/ hit index the original hit id stored on file.
+    std::string outputMeasurementIds;
+    /// Output hit-particles mapping collection.
+    std::string outputMeasurementParticlesMap;
+    /// Output simulated (truth) hits collection.
+    std::string outputSimHits;
+    /// Tracking geometry required to access global-to-local transforms.
+    std::shared_ptr<const Acts::TrackingGeometry> trackingGeometry;
```

Comment:
Same question as for the writer: the measurement information is purely local, why are the transformation required?

Reply:
Removed now.